### PR TITLE
Convert cfgmanifest linux entries to matching format

### LINKED ISSUE #1838: Use the "linux" component type within the cgmanifest whenever appropriate (e.g. clang-format)
For example, the following is currently marked as "other":
https://github.com/Azure/azure-sdk-for-c/blob/75a4910e7e034949897e9cc2cb88f4a168391f3d/cgmanifest.json#L86-L96

https://docs.opensource.microsoft.com/tools/cg/features/cgmanifest/
image

> The component type defined the package type it can be npm, nuget, maven, etc ... in this case this is an ubuntu package for what I can see, when the packages are from ubuntu, debian alpine or other linux distribution we recommend to use the component type "linux"

cc @RickWinter, @jcfiorenzano

### PR DESCRIPTION
**Pull request by antkmsft**
Closes #1838.

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -64,32 +64,35 @@
     {
       "Component": {
         "Type": "linux",
-        "Other": {
+        "linux": {
```
Comment (ahsonkhan) [inline cgmanifest.json:67]: Need to fix `gcc-arm-none-eabi` as well, above.

File: cgmanifest.json [hunk 2]
```diff
@@ -64,32 +64,35 @@
...
           "Name": "clang-format",
           "Version": "9.0.0-2",
-          "DownloadUrl": "https://ubuntu.pkgs.org/18.04/ubuntu-updates-universe-amd64/clang-format-9_9-2~ubuntu18.04.2_amd64.deb.html"
+          "Distribution": "Ubuntu",
+          "Release": "18.04"
```
Comment (ahsonkhan) [inline cgmanifest.json:95]: I wonder if we need to specify `18.04.2` for this one, or if cg finds the package correctly from the distribution. We may need to test this out.

File: cgmanifest.json [hunk 3]
```diff
@@ -64,32 +64,35 @@
     {
       "Component": {
         "Type": "linux",
-        "Other": {
+        "linux": {
           "Name": "gcovr",
           "Version": "3.4-1",
-          "DownloadUrl": "https://ubuntu.pkgs.org/18.04/ubuntu-universe-amd64/gcovr_3.4-1_all.deb.html"
+          "Distribution": "Ubuntu",
+          "Release": "18.04"
```
Comment (ahsonkhan) [inline cgmanifest.json:71]: I think we may also need Pool-URL and/or Key-URL.

Maybe follow up with cg folks or look up other repos that use the `linux` component type, to see what is the full amount of detail we need to provide.

> Valid values for distribution: Debian, Alpine, Rhel, Centos, Fedora, Ubuntu. Pool-URL Example (http://archive.ubuntu.com/ubuntu) and Key-URL Example (http://archive.ubuntu.com/ubuntu/project/ubuntu-archive-keyring.gpg)

**Review by RickWinter — APPROVED** [review]

**Review by ahsonkhan — APPROVED** [review]

File: cgmanifest.json (hunk 3 above)
Comment (jcfiorenzano) [reply cgmanifest.json]: Yes this linux dependency looks correct

File: cgmanifest.json (hunk 2 above)
Comment (jcfiorenzano) [reply cgmanifest.json]: I think that if you can specify 18.04.2 is better

File: cgmanifest.json (hunk 3 above)
Comment (ahsonkhan) [reply cgmanifest.json]: Where do we typically find "valid" values for Pool and Key URLs for the different distros?

**Pull request merged**